tests/integration/ie-tests.sh

```diff
@@ -174,6 +174,157 @@ cleanup_user_failure() {
+  kubectl exec testpod -- env KCONFIG="$MY_KUBE_CONFIG" /bin/bash -c 'echo "$KCONFIG" > /home/spark/.kube/config'
+  kubectl exec testpod -- /bin/bash -c 'cat /home/spark/.kube/config'
+}
+
+teardown_test_pod() {
+  kubectl delete pod testpod
+}
+
+run_example_job_in_pod() {
+  SPARK_EXAMPLES_JAR_NAME='spark-examples_2.12-3.3.2.jar'
+
+  PREVIOUS_JOB=$(kubectl get pods | grep driver | tail -n 1 | cut -d' ' -f1)
+
+  NAMESPACE=$1
+  USERNAME=$2
+
+  kubectl exec testpod -- env UU="$USERNAME" NN="$NAMESPACE" JJ="$SPARK_EXAMPLES_JAR_NAME" \
+                  /bin/bash -c 'spark-client.spark-submit \
+                  --username $UU --namespace $NN \
+                  --conf spark.kubernetes.driver.request.cores=100m \
+                  --conf spark.kubernetes.executor.request.cores=100m \
+                  --class org.apache.spark.examples.SparkPi \
+                  local:///opt/spark/examples/jars/$JJ 100'
+
+  # kubectl --kubeconfig=${KUBE_CONFIG} get pods
+  DRIVER_JOB=$(kubectl get pods -n ${NAMESPACE} | grep driver | tail -n 1 | cut -d' ' -f1)
+
+  if [[ "${DRIVER_JOB}" == "${PREVIOUS_JOB}" ]]
+  then
+    echo "ERROR: Sample job has not run!"
+    exit 1
+  fi
+
+  # Check job output
+  pi=$(kubectl logs $(kubectl get pods -n ${NAMESPACE} | grep driver | tail -n 1 | cut -d' ' -f1)  -n ${NAMESPACE} | grep 'Pi is roughly' | rev | cut -d' ' -f1 | rev | cut -c 1-3)
+  echo -e "Spark Pi Job Output: \n ${pi}"
+
+  if [ "${pi}" != "3.1" ]; then
+      exit 1
+  fi
+
+}
+
+test_example_job_in_pod() {
+  run_example_job_in_pod tests spark
+}
+
+
+run_spark_shell_in_pod() {
+  echo "run_spark_shell_in_pod ${1} ${2}"
+
+  NAMESPACE=$1
+  USERNAME=$2
+
+  echo "import scala.math.random" > test-spark-shell.scala
+  echo "val slices = 10" >> test-spark-shell.scala
+  echo "val n = math.min(100000L * slices, Int.MaxValue).toInt" >> test-spark-shell.scala
+  echo "val count = spark.sparkContext.parallelize(1 until n, slices).map { i => val x = random * 2 - 1; val y = random * 2 - 1;  if (x*x + y*y <= 1) 1 else 0;}.reduce(_ + _)" >> test-spark-shell.scala
+  echo "println(s\"Pi is roughly \${4.0 * count / (n - 1)}\")" >> test-spark-shell.scala
+  echo "System.exit(0)" >> test-spark-shell.scala
+
+  SPARK_SHELL_COMMANDS=$(cat ./test-spark-shell.scala)
+
+  echo -e "$(kubectl exec testpod -- env UU="$USERNAME" NN="$NAMESPACE" CMDS="$SPARK_SHELL_COMMANDS" /bin/bash -c 'echo "$CMDS" | spark-client.spark-shell --username $UU --namespace $NN')" > spark-shell.out
+
+  pi=$(cat spark-shell.out  | grep "^Pi is roughly" | rev | cut -d' ' -f1 | rev | cut -c 1-3)
+  echo -e "Spark-shell Pi Job Output: \n ${pi}"
+  rm spark-shell.out test-spark-shell.scala
+  if [ "${pi}" != "3.1" ]; then
+      exit 1
+  fi
+}
+
+test_spark_shell_in_pod() {
+  run_spark_shell_in_pod tests spark
+}
+
+run_pyspark_in_pod() {
+  echo "run_pyspark_in_pod ${1} ${2}"
+
+  NAMESPACE=$1
+  USERNAME=$2
+
+  echo "import sys" > test-pyspark.py
```

Comment:
Is it possible to create a *.py file and  put it into the test folder, instead of doing it with the following commands ?